### PR TITLE
MM-33818: Add replica lag metric

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://github.com/mattermost/mattermost-server/pull/17148

Updated:
- Self-Managed Admin Guide > Scaling Mattermost > Performance Monitoring (E20) > Getting Started > Database Metrics
   - Added 2 new metrics: ``mattermost_db_replica_lag_abs`` and ``mattermost_db_replica_lag_time``

### CODE REVIEW

File: source/administration/config-settings.rst [hunk 1]
```diff
@@ -4863,6 +4863,43 @@ Changes to this setting require a server restart before taking effect.
...
+
+*Available in Enterprise Edition E20*
+
+Specifies a connection string and user-defined SQL queries on the database to measure replica lag for a single replica instance. These settings monitor absolute lag based on binlog distance/transaction queue length, and the time taken for the replica to catch up.
+
++-------------------------------------------------------------------------------------------------------+
+| This feature’s ``config.json`` setting is ``"ReplicaLagSettings": []`` with string array input.       |
++-------------------------------------------------------------------------------------------------------+
+
+String array input can consist of:
```
Comment (agnivade) [inline source/administration/config-settings.rst:4877]: Perhaps the word "can" implies that other input can also be there? Maybe it's good to just say: "String array input consists of:"

File: source/administration/config-settings.rst [hunk 2]
```diff
@@ -4863,6 +4863,43 @@ Changes to this setting require a server restart before taking effect.
...
+
+- ``DataSource`` - The DB credentials to connect to the replica instance.
+- ``QueryAbsoluteLag`` - A plain SQL query that must return a single row. The first column must be the node value of the Prometheus metric, and the second |column must be the value of the lag used to measure absolute lag.
```
Comment (agnivade) [inline source/administration/config-settings.rst:4880]: There is a `|` before column

**Review by agnivade — APPROVED** [review]

File: source/administration/config-settings.rst (hunk 1 above)
Comment (cwarnermm) [inline source/administration/config-settings.rst:4877]:
```suggestion
String array input consists of:
```

File: source/administration/config-settings.rst (hunk 2 above)
Comment (cwarnermm) [inline source/administration/config-settings.rst:4880]:
```suggestion
- ``QueryAbsoluteLag`` - A plain SQL query that must return a single row. The first column must be the node value of the Prometheus metric, and the second column must be the value of the lag used to measure absolute lag.
```

**Comment by cwarnermm** [comment]
Thanks, @agnivade!

**Review by adamjclarkson — APPROVED** [review]

File: source/deployment/metrics.rst [hunk 3]
```diff
@@ -149,6 +149,8 @@ Database Metrics
 - ``mattermost_db_read_replica_connections_total``: The total number of connections to all the read replica databases.
 - ``mattermost_db_search_replica_connections_total``: The total number of connections to all the search replica databases.
 - ``mattermost_db_store_time``: The total time in seconds to execute a given database store method.
+- ``mattermost_db_replica_lag_abs``: - Absolute lag time based on binlog distance/transaction queue length.
```
Comment (justinegeffen) [inline source/deployment/metrics.rst:152]:
```suggestion
- ``mattermost_db_replica_lag_abs``: Absolute lag time based on binlog distance/transaction queue length.
```

File: source/deployment/metrics.rst [hunk 4]
```diff
@@ -149,6 +149,8 @@ Database Metrics
...
+- ``mattermost_db_replica_lag_time``: - The time taken for the replica to catch up.
```
Comment (justinegeffen) [inline source/deployment/metrics.rst:153]:
```suggestion
- ``mattermost_db_replica_lag_time``: The time taken for the replica to catch up.
```

File: source/administration/config-settings.rst [hunk 5]
```diff
@@ -4863,6 +4863,43 @@ Changes to this setting require a server restart before taking effect.
...
+
+Specifies a connection string and user-defined SQL queries on the database to measure replica lag for a single replica instance. These settings monitor absolute lag based on binlog distance/transaction queue length, and the time taken for the replica to catch up.
+
++-------------------------------------------------------------------------------------------------------+
+| This feature’s ``config.json`` setting is ``"ReplicaLagSettings": []`` with string array input.       |
++-------------------------------------------------------------------------------------------------------+
+
+String array input consists of:
+
+- ``DataSource`` - The DB credentials to connect to the replica instance.
```
Comment (justinegeffen) [inline source/administration/config-settings.rst:4879]:
```suggestion
- ``DataSource``: The DB credentials to connect to the replica instance.
```

File: source/administration/config-settings.rst [hunk 6]
```diff
@@ -4863,6 +4863,43 @@ Changes to this setting require a server restart before taking effect.
...
+- ``QueryAbsoluteLag`` - A plain SQL query that must return a single row. The first column must be the node value of the Prometheus metric, and the second column must be the value of the lag used to measure absolute lag.
```
Comment (justinegeffen) [inline source/administration/config-settings.rst:4880]:
```suggestion
- ``QueryAbsoluteLag``: A plain SQL query that must return a single row. The first column must be the node value of the Prometheus metric, and the second column must be the value of the lag used to measure absolute lag.
```

File: source/administration/config-settings.rst [hunk 7]
```diff
@@ -4863,6 +4863,43 @@ Changes to this setting require a server restart before taking effect.
...
+- ``QueryTimeLag`` - A plain SQL query that must return a single row. The first column must be the node value of the Prometheus metric, and the second column must be the value of the lag used to measure the time lag.
```
Comment (justinegeffen) [inline source/administration/config-settings.rst:4881]:
```suggestion
- ``QueryTimeLag``: A plain SQL query that must return a single row. The first column must be the node value of the Prometheus metric, and the second column must be the value of the lag used to measure the time lag.
```

**Review by justinegeffen — COMMENTED** [review]
Minor suggestions prior to merge. :)

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**